apps/homescreen/js/app.js

```diff
@@ -321,6 +324,21 @@ const SETTINGS_VERSION = 0;
   }
 
   App.prototype = {
+    get iconSize() {
+      // If this._iconSize is undefined or 0, let's refresh the value.
+      if (!this._iconSize) {
+        for (var i = 0; i < this.icons.childNodes.length; i++) {
```

Comment:
This loop will be very slow, better to do:
var children = this.icons.children;
for (var i = 0, iLen = children.length; i < iLen; i++) { ... }

Because the child-list in gaia-container is virtual, it gets built on every access of it.